scripts/nb-tester/test-notebook.py

```diff
@@ -168,6 +189,9 @@ def _execute_notebook(filepath: Path, args: argparse.Namespace) -> nbformat.Note
     if not args.write:
         return nb
 
+    if not submit_jobs:
+        remove_mocking_cell(nb)
```

Comment:
One option is to use a context manager, like a `with` statement. While you're in the `with` block, it will patch the notebook. Once you exit it, we remove the patch. It might make the code more readable? 

You'd still keep the dedicated functions you have, only define a new context manager like https://www.pythonmorsels.com/creating-a-context-manager/

Reply:
Good idea, added in ddb05c97d334ac789288e9efac91c3995a033b64. Thanks!